tools/ClassGen/NodeBuilder.cpp

```diff
@@ -352,23 +405,53 @@ void NodeBuilder::emitPrettyPrinter(std::ostream &os) const {
   }
 
   // Generate description for outputs.
+  // --> Fix outputs.
   for (const auto &op : nodeOutputs_) {
     os << "  db.addParam(\"" << op.second << "\", *(get" << op.second
        << "().getType()));\n";
   }
+  // --> variable outputs.
+  if (variableOutputMembers_.size()) {
+    os << "  unsigned_t variableOutIdx = " << nodeOutputs_.size() << ";\n";
+    for (const auto &op : variableOutputMembers_) {
+      os << "  for (unsigned i = 0; i < " << op.second << "_size_; i++) {\n"
+         << "   db.addParam(\"" << op.second << "\" + std::to_string(i),"
+         << " *(getNthResult(variableOutIdx + i).getType()));\n"
+         << "  }\n"
+         << "  variableOutIdx += " << op.second << "_size_;\n";
+    }
+  }
 
   os << "  return db;\n}\n";
 }
 
 void NodeBuilder::emitCloner(std::ostream &os) const {
   os << "\nNode* " << name_ << "Node::clone() const {\n";
 
+  // Variable outputs
+  if (variableOutputMembers_.size()) {
+    os << "  unsigned_t variableOutIdx = " << nodeOutputs_.size() << ";\n";
+    for (const auto &op : variableOutputMembers_) {
+      os << "  std::vector<TypeRef> ResultTypes_of_" << op.second << ";\n"
```

Comment:
We usually don't use underscores.

Reply:
Nit: Since Node has "types_" vector, couldn't you just use an ArrayRef here which would just be a slice of the "types_" array corresponding to the variable output? This way you don't need to copy anything around.

Or do we need to pass it as a  vector later?

Reply:
I think `ArrayRef `is fine. `Node` class `type_` member is protected we we may indeed use `ArrayRef` to avoid creating vectors on the fly.

Reply:
Removed `_` to stick the the Glow naming convention.
Also used `ArrayRef` slices to avoid building vectors on the fly.